### PR TITLE
Update to Angular 2 RC1

### PR DESCRIPTION
**Pull request by Shawn-Fan**
Updated app.component.ts
               basemapselect.component.ts
               boot.ts
               layer.component.ts
               legend.component.ts
               map.component.ts
               map.service.ts
               search.componet.ts
               index.html
               package.json
added systemjs.config.js

### CODE REVIEW

**Comment by Shawn-Fan** [comment]
it works well in Angular 2 RC1, but  css issue in "search" component

File: index.html [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
+    <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"> 
     <link rel="stylesheet" href="//js.arcgis.com/3.16/esri/themes/calcite/dijit/calcite.css">
     <link rel="stylesheet" href="//js.arcgis.com/3.16/esri/themes/calcite/esri/esri.css">
     <link rel="stylesheet" href="app/styles/main.css">
+
+    
     <title>ArcGIS JS API 3 & Angular 2</title>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="styles.css">
```
Comment (tomwayson) [inline index.html:11]: There is no styles.css, this should be removed.

File: index.html [hunk 2]
```diff
@@ -21,39 +27,29 @@
     <script src="https://js.arcgis.com/3.16/"></script>
     <script src="node_modules/esri-system-js/dist/esriSystem.js"></script>
   </head>
-
+  
   <!-- 3. Display the application -->
-  <body class="calcite">
```
Comment (tomwayson) [inline index.html:26]: removing this class from the body is probably what's causing the issues w/ the styles on the Search widget

File: index.html [hunk 3]
```diff
@@ -21,39 +27,29 @@
...
-    <div class="container-fluid">
```
Comment (tomwayson) [inline index.html:27]: is there a reason to remove this container div?

File: package.json [hunk 4]
```diff
@@ -19,18 +19,41 @@
...
   "devDependencies": {
-    "concurrently": "^1.0.0",
-    "lite-server": "^1.3.1",
-    "typescript": "^1.7.3"
+    "concurrently": "^2.0.0",
+    "lite-server": "^2.2.0",
+    "typescript": "^1.8.10",
+    "typings": "^0.8.1",
+
+    "canonical-path": "0.0.2",
```
Comment (tomwayson) [inline package.json:46]: Are any of these extra devDependencies (from cannonical-path down) necessary?

File: index.html [hunk 5]
```diff
@@ -21,39 +27,29 @@
...
-      System.import('app/boot')
-        .then(null, console.error.bind(console));
-    }, {
-      moduleNameOverrides: {
-        'esri/arcgis/utils': 'arcgisUtils',
-        'esri/basemaps': 'esriBasemaps'
-      }
-    });
-  </script>
+      // load esri modules needed by this application
```
Comment (tomwayson) [inline index.html:36]: This should probably be moved into system.config.js now. That would also require it's script tag to be moved below the script tag for esriSystem.js

**Comment by tomwayson** [comment]
@Shawn-Fan 

This is great! I was able to get the app up and running on RC.1 after pulling your branch and running;

``` bash
rm -rf node_modules
npm install
```

I've made a few comments above with minor changes that should help fix some style issues (including the Search widget) and remove some console errors.

However, In addition to the above fixes, I now see many errors when I run `npm test` (which is just calling `tsc`):

``` bash
> angular2-esri-example@0.0.2 test /Users/tom6947/code/angular2-esri-example
> tsc

app/map.service.ts(2,59): error TS2307: Cannot find module 'esri'.
node_modules/@angular/common/src/directives/ng_class.d.ts(72,35): error TS2304: Cannot find name 'Set'.
node_modules/@angular/common/src/facade/async.d.ts(27,33): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/facade/async.d.ts(28,45): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/facade/lang.d.ts(4,17): error TS2304: Cannot find name 'Map'.
node_modules/@angular/common/src/facade/lang.d.ts(5,17): error TS2304: Cannot find name 'Set'.
node_modules/@angular/common/src/facade/lang.d.ts(70,59): error TS2304: Cannot find name 'Map'.
node_modules/@angular/common/src/facade/promise.d.ts(2,14): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/facade/promise.d.ts(8,32): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/facade/promise.d.ts(9,38): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/facade/promise.d.ts(10,35): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/facade/promise.d.ts(10,93): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/facade/promise.d.ts(11,34): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/facade/promise.d.ts(11,50): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/facade/promise.d.ts(12,32): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/facade/promise.d.ts(12,149): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/facade/promise.d.ts(13,43): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/forms/directives/select_control_value_accessor.d.ts(18,17): error TS2304: Cannot find name 'Map'.
node_modules/@angular/common/src/pipes/async_pipe.d.ts(29,29): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/pipes/async_pipe.d.ts(35,38): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/pipes/async_pipe.d.ts(37,39): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/common/src/pipes/async_pipe.d.ts(39,44): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/application_ref.d.ts(39,88): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/application_ref.d.ts(99,42): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/application_ref.d.ts(174,33): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/change_detection/differs/default_keyvalue_differ.d.ts(24,15): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/change_detection/differs/default_keyvalue_differ.d.ts(26,16): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/di/reflective_provider.d.ts(105,123): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/di/reflective_provider.d.ts(105,165): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/async.d.ts(27,33): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/async.d.ts(28,45): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/collection.d.ts(1,25): error TS2304: Cannot find name 'MapConstructor'.
node_modules/@angular/core/src/facade/collection.d.ts(2,25): error TS2304: Cannot find name 'SetConstructor'.
node_modules/@angular/core/src/facade/collection.d.ts(4,27): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(4,39): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(7,9): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(8,30): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(11,43): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(12,27): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(14,23): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(15,25): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/collection.d.ts(100,41): error TS2304: Cannot find name 'Set'.
node_modules/@angular/core/src/facade/collection.d.ts(101,22): error TS2304: Cannot find name 'Set'.
node_modules/@angular/core/src/facade/collection.d.ts(102,25): error TS2304: Cannot find name 'Set'.
node_modules/@angular/core/src/facade/lang.d.ts(4,17): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/lang.d.ts(5,17): error TS2304: Cannot find name 'Set'.
node_modules/@angular/core/src/facade/lang.d.ts(70,59): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/facade/promise.d.ts(2,14): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(8,32): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(9,38): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(10,35): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(10,93): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(11,34): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(11,50): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(12,32): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(12,149): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/facade/promise.d.ts(13,43): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/component_resolver.d.ts(8,53): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/component_resolver.d.ts(12,44): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/dynamic_component_loader.d.ts(62,148): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/dynamic_component_loader.d.ts(103,144): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/dynamic_component_loader.d.ts(108,139): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/linker/dynamic_component_loader.d.ts(109,135): error TS2304: Cannot find name 'Promise'.
node_modules/@angular/core/src/reflection/reflector.d.ts(28,22): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/reflection/reflector.d.ts(30,15): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/reflection/reflector.d.ts(32,15): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/reflection/reflector.d.ts(34,15): error TS2304: Cannot find name 'Map'.
node_modules/@angular/core/src/reflection/reflector.d.ts(36,16): error TS2304: Cannot find name 'Set'.
node_modules/@angular/core/src/testability/testability.d.ts(40,20): error TS2304: Cannot find name 'Map'.
node_modules/@angular/platform-browser-dynamic/platform_browser_dynamic.d.ts(75,90): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/Observable.d.ts(10,66): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/Observable.d.ts(66,60): error TS2304: Cannot find name 'Promise'.
node_modules/rxjs/Observable.d.ts(66,70): error TS2304: Cannot find name 'Promise'.
npm ERR! Test failed.  See above for more details.
```

After looking into the [typings section of the Angular 2 TS config docs](https://angular.io/docs/ts/latest/guide/typescript-configuration.html#!#typings) I realized it was b/c we don't have the typings for `core-js` installed. I had to run the following commands to get that working:

``` bash
rm -rf typings/
npm install --save-dev typings@">=1.0.4"
npm run typings -- install dt~core-js --save --global
```

After that I only see the one [known and expected error](https://github.com/Esri/esri-system-js/issues/6) when running `npm test`:

``` bash
> angular2-esri-example@0.0.2 test /Users/tom6947/code/angular2-esri-example
> tsc

app/map.service.ts(2,59): error TS2307: Cannot find module 'esri'.
```

Please let me know if you are interested in making the above changes yourself and adding them as follow on commit(s) to this pull request, or if you would prefer that I just handle them.

**Comment by Shawn-Fan** [comment]
I am going to try to figure it out and fix it

________________________________

**Comment by Shawn-Fan** [comment]
Hi Tom
Thank you so much for your info and correction. I did some change but I am not sure about "This should probably be moved into system.config.js now. That would also require it's script tag to be moved below the script tag for esriSystem.js", Can you make the final correction?

**Comment by Shawn-Fan** [comment]
I updated 2 files above, removed line 12 but I don't know why it is lin

**Comment by tomwayson** [comment]
@Shawn-Fan I see that you've closed this. Is that b/c you no longer intend for it to be merged?

**Comment by tomwayson** [comment]
I'm re-opening assuming that  you closed it on accident.

**Comment by Shawn-Fan** [comment]
Honestly I didn't close it.  l'll try to update it because there is "critical condition" in my computer system recently.

**Comment by tomwayson** [comment]
No problem. I'm just about finished merging this.

**Comment by tomwayson** [comment]
All merged, thank you @Shawn-Fan!

**Comment by tomwayson** [comment]
@Shawn-Fan, also be sure to pull the latest master and run `npm install` to get the latest changes as I had to make a few changes on top of your commits.

**Comment by Shawn-Fan** [comment]
Thanks, it works perfect!

**Pull request merged**